src/asset-descriptions/asset-descriptions.service.ts

```diff
@@ -50,4 +57,65 @@ export class AssetDescriptionsService {
       },
     });
   }
+
+  async list(options: ListAssetDescriptionsOptions): Promise<{
+    data: AssetDescription[];
+    hasNext: boolean;
+    hasPrevious: boolean;
+  }> {
+    const cursorId = options.before ?? options.after;
+    const cursor = cursorId ? { id: cursorId } : undefined;
+    const direction = options.before !== undefined ? -1 : 1;
+    const limit =
+      direction * Math.min(MAX_LIMIT, options.limit || DEFAULT_LIMIT);
+    const orderBy = { id: SortOrder.DESC };
+    const skip = cursor ? 1 : 0;
+    const where = {
+      asset_id: options.assetId,
+    };
+    const data = await this.prisma.assetDescription.findMany({
+      cursor,
+      orderBy,
+      skip,
+      take: limit,
+      where,
+    });
+
```

Comment:
Can we dump this method below and just use a query to get the metadata like

`select min(id), max(id) where ${where} order by ${orderBy}`

Then you can just look to make sure the min < firstItem and max is > lastItem.id


Took me a bit to understand what was happening below with the method, it is doing something pretty simple but with some pretty sophisticated tools (`findMany` and `AssetDescription[]`)

Reply:
Up to you if you want to do this though

Reply:
Yeah, let me look into this for a subsequent PR. We're doing this for pagination across the API, but I'll see if there's a simpler pattern